src/main/java/org/templegalaxia/patterns/acjs/Spoken.java

```diff
@@ -0,0 +1,216 @@
+          .setDescription("Number of lights to use (density).");
+
+  public final CompoundParameter birthRate =
+      new CompoundParameter("Rate", 1.0, 0, 1)
+          .setDescription("How likely a new name is to appear (per second).");
+
+  public final CompoundParameter dot =
+      (CompoundParameter)
+          new CompoundParameter("Dot speed", 1000, 0, 10000)
+              .setUnits(LXParameter.Units.MILLISECONDS)
+              .setDescription("Length of a 'dot' in morse code.");
+
+  public final CompoundParameter dotVariance =
+      (CompoundParameter)
+          new CompoundParameter("Dot variance", 0, 0, 1000)
+              .setUnits(LXParameter.Units.MILLISECONDS)
+              .setDescription("How variable dot length will be between names.");
+
+  private final HashMap<Character, String> morseLookup = new HashMap<>();
+  private final List<String> names = new ArrayList<>();
+  private final List<MorsePulser> pulsers = new ArrayList<>();
+  private final List<Integer> availableLights = new ArrayList<>();
+
+  private final Random rand = new Random();
+
+  public Spoken(LX lx) {
+    super(lx);
+    addParameter(density);
+    addParameter(birthRate);
+    addParameter(dot);
+    addParameter(dotVariance);
+
+    loadMorse();
+
+    for (int i = 0; i < colors.length; i++) availableLights.add(i);
+  }
+
+  void loadMorse() {
+    // Shamelessly boosted from the model load code.
+    CSVParser parser;
+    try {
+      parser =
+          new CSVParser(
+              new BufferedReader(
+                  new InputStreamReader(Spoken.class.getResourceAsStream("morse.csv"))),
+              CSVFormat.EXCEL.withHeader());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    for (CSVRecord csvRecord : parser) {
+      morseLookup.put(csvRecord.get("Letter").charAt(0), csvRecord.get("Morse"));
+    }
+  }
+
+  void loadNames() {
+    Path dataPath = Paths.get(NAME_DATA);
+
+    if (Files.exists(dataPath) && Files.isRegularFile(dataPath)) {
+      try (Stream<String> nameLines = Files.lines(dataPath)) {
+        nameLines.forEach(names::add);
+      } catch (IOException e) {
+        e.printStackTrace();
+      } finally {
+        // Backup / default
+        names.add("Larry Harvey");
+      }
+    }
+
+    Collections.shuffle(names);
+  }
+
+  public void newPulser(String message) {
+    double dotLength = dot.getValue();
+    dotLength += (Math.random() - 0.5) * dotVariance.getValue();
+
+    Collections.shuffle(availableLights);
+    int light = availableLights.remove(0);
+
+    MorsePulser pulser = new MorsePulser(lx, light, message, dotLength);
+    pulsers.add(pulser);
+    addLayer(pulser);
+  }
+
+  public void run(double deltaMs) {
+    // Give each unlit 'slot' a chance to be born.
+    for (int i = 0; i < (int) density.getValue() - pulsers.size(); i++) {
+      if (names.size() == 0) loadNames(); // Reload names if we're out. XXX: Performance?
```

Comment:
Yeah, this definitely feels off. Reloading the names is going to give you the same list every time (just shuffled differently). I think it makes more sense to just shuffle it once on initial load and then loop through them. Or you can periodically shuffle again if you don't like the idea of it being in the same loop-order.

Reply:
The idea here was that we could update the underlying file during run with new names, captured through some kind of TBD on-playa means, and that they would get loaded in. It definitely doesn't need to be reloaded every time the list is exhausted though.

If you could give me an idea of the lighting setup on-playa (a headless machine running the same LX instance all burn versus different laptops/instances being switched in and out, for instance), happy to come up with a better solution to hot reloading. If you don't think it's practical to update at all, I can take it out.

Reply:
@antonjs It's going to be on a headless linux machine in a NEMA enclosure at the base of the temple, and I believe we'll also have a wireless access point on it. If you want to try and devise some means of getting user input, we could probably set you up with an account on that box with permissions to transfer files to a specific place on that box.

Let us know if you work out more specific plans about how to collect names, but be advised that over the next few days a lot of us will be to playa and might be harder to reach after that.

Reply:
Thanks @deleted. Current thinking is an retro terminal that would allow user input, located somewhere that works for you all / the broader temple crew. Uploading a text file over wifi would definitely work. I can put together a more detailed plan and get it to you asap.

I'll be on playa from 22 Aug (Wednesday) so can coordinate in person then, assuming of course you don't have more urgent chaos to attend to...

Reply:
That sounds very cool. Keep us posted!

I've noticed that @alanperpersak has also submitted a pattern that [encodes and displays string messages](https://github.com/temple2018/Galaxia/pull/32/files#diff-dbd65fe961ae37d61153c255d1e0f413). Perhaps the two of you would like to coordinate on input sources. I'll drop a note in his PR as well.

Reply:
Thanks @deleted, I'll drop Alan a line. I've added a (slightly) more fleshed out proposal for doing the collecting names below, please let me know what you think!